src/VariantViewer.js

```diff
@@ -31,10 +31,23 @@ var variantsFill = function(d, fv) {
         } else {
             return LegendDialog.UPNonDiseaseColor;
         }
-    } else if (d.siftScore !== undefined) {
-        return LegendDialog.getPredictionColor((d.siftScore + (1-d.polyphenScore))/2);
     } else {
-        return LegendDialog.othersColor;
+        var sift = false, polyphen = false;
```

Comment:
On 29/04/2016 15:39, Xavier wrote:
>
> In src/VariantViewer.js 
> <https://github.com/ebi-uniprot/biojs-vis-proteinFeaturesViewer/pull/11#discussion_r61587704>:
>
> >      } else {
> > -        return LegendDialog.othersColor;
> > +        var sift = false, polyphen = false;
>
> I think this could be done in less lines of code. You don't actually 
> use polyphenPrediction/siftPrediction just siftScore/polyphenScore so 
> should be able to check they exist directly in the conditionals.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/ebi-uniprot/biojs-vis-proteinFeaturesViewer/pull/11/files/10734263453714ee99d113b315545325c9b0b2da#r61587704>
>
I agree, but this one is tricky. Sometimes we have no prediction but 
still we have scores, so I tried to cover everything is there.